File: src/app/api/send-welcome-email/route.rs
import { NextResponse } from "next/server";
import { sendWelcomeEmail } from "@/lib/email/send";

export async function POST(req: Request) {
  try {
    const body = await req.json();
    const { to, memberName, churchName, tempPassword } = body;

    if (!to || !memberName || !churchName || !tempPassword) {
      return NextResponse.json(
        { error: "Missing required fields" },
        { status: 400 }
      );
    }

    const result = await sendWelcomeEmail({
      to,
      memberName,
      churchName,
      tempPassword,
    });

    return NextResponse.json({ success: true, result });
  } catch (error) {
    console.error("API send-welcome-email error:", error);
    return NextResponse.json(
      { error: "Failed to send email" },
      { status: 500 }
    );
  }
}

File: src/app/api/auth/session/route.ts
import { NextResponse } from "next/server";
import { getSessionFromCookieHeader } from "@/lib/auth/server-session";

export async function GET(req: Request) {
  const session = getSessionFromCookieHeader(req.headers.get("cookie"));
  if (!session) {
    return NextResponse.json({ authenticated: false }, { status: 401 });
  }

  return NextResponse.json({ authenticated: true, session });
}
